From 407cce8dd8b0e621449c6c3015c303539f53819b Mon Sep 17 00:00:00 2001
From: adisbladis <adisbladis@gmail.com>
Date: Fri, 28 Dec 2018 16:54:34 +0000
Subject: [PATCH] Fix startup using non-posix shells

* xcb.el (xcb:create-auth-info): Use lisp for string substitution instead of shell
---
 xcb.el | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/xcb.el b/xcb.el
index 7ed2d3e..a769d2f 100644
--- a/xcb.el
+++ b/xcb.el
@@ -147,7 +147,11 @@
 (defun xcb:create-auth-info ()
   "Create the default `auth-info'."
   (let ((xauth-output (shell-command-to-string
-                       "xauth list ${DISPLAY#localhost} 2>/dev/null"))
+                       (concat
+                        "xauth list "
+                        (replace-regexp-in-string "^localhost" ""
+                                                  (getenv "DISPLAY"))
+                        " 2>/dev/null")))
         (name "MIT-MAGIC-COOKIE-1") ;only support MIT-MAGIC-COOKIE-1 protocol.
         (data ""))
     (if (string= "" xauth-output)
-- 
2.19.2

